### PR TITLE
Implementation of Card part bar histogram

### PR DESCRIPTION
**Pull request by badrinathvm**
histogram

## Issue Link :link:
 <ul>
   <li> New Feature </li>
   <li> None </li>
</ul>

## Goals of this PR :tada:
 <ul>
   <li> Enhancement </li>
   <li> What does this fix? </li>
   <li> Yes </li>
 </ul>
 
## How Has This Been Tested :mag:

Please let us know if you have tested your PR and if we need to reproduce the issues. Also, please let us know if we need any relevant information for running the tests.

<ul>
 <li> User Interface Testing </li>
 <li> Application Testing </li>
</ul>

## Test Configuration :space_invader:

<ul>
 <li> Xcode version: 11.1 </li>
 <li> Device/Simulator - iPhone 11 </li>
 <li> iOS version - 13.1 </li>
</ul>

## Things to check on :dart:

 - [ ] My Pull Request code follows the coding standards and styles of the project 
 - [ ] I have worked on unit tests and reviewed my code to the best of my ability 
 - [ ] I have used comments to make other coders understand my code better 
 - [ ] My changes are good to go without any warnings 
 - [ ] I have added unit tests both for the happy and sad path 
 - [ ] All of my unit tests pass successfully before pushing the PR 
 - [ ] I have made sure all dependent downstream changes impacted by my PR are working

### CODE REVIEW

**Comment by lwdupont** [comment]
Wow neat! I'll have a look tonight.

**Comment by lwdupont** [comment]
(unless Chase beats me to it ;) )

File: CardParts/src/Classes/Card Parts/CardPartHistogramView.swift [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        }
+    }
+
+    /// shows all three lines of the bar graph
+    public var histogramLines:HistogramLine = HistogramLine.lines(bottom: true, middle: true, top: true)
+    
+    /// Responsible for compute all positions and frames of all elements represent on the bar chart
+    private var presenter:CardPartBarChart = CardPartBarChart(barWidth: 20, space: 10)
+    
+    /// height for the histiogram to render
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartHistogramView.swift]: Spelling error

File: CardParts/src/Classes/Card Parts/CardPartHistogramView.swift [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+          self.layer.sublayers?.forEach({
+              if $0 is CAShapeLayer {
+                  $0.removeFromSuperlayer()
+              }
+          })
+        
+        /// contains horizontal bottom, middle and top lines
+        let lines = presenter.computeHorizontalLines(viewHeight: self.frame.height)
+
+        if case let HistogramLine.lines(bottom,middle,top) = histogramLines {
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartHistogramView.swift]: Nice!

File: CardParts/src/Classes/Card Parts/CardPartHistogramView.swift [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    private var linePadding:CGFloat = 6.5
+    
+    var dataEntries: [DataEntry] = []
+       
+    init(barWidth:CGFloat, space: CGFloat) {
+        self.barWidth = barWidth
+        self.space = space
+    }
+    
+    /// width of the entier bars
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartHistogramView.swift]: Spelling

File: CardParts/src/Classes/Card Parts/CardPartTriangleView.swift [hunk 4]
```diff
@@ -7,7 +7,10 @@
 
 import Foundation
 
-class CardPartTriangleView: UIView {
+class CardPartTriangleView: UIView,CardPartView {
+        
+    var margins: UIEdgeInsets = CardParts.theme.cardPartMargins
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartTriangleView.swift]: What is this needed for?

File: README.md [hunk 5]
```diff
@@ -988,6 +989,22 @@ Provides the capability to add  radio buttons with configurable inner/outer circ
...
+```swift
+    let dataEntries = self.generateRandomDataEntries()
+    barHistogram.width = 8
+    barHistogram.spacing = 8
+    barHistogram.histogramLines = HistogramLine.lines(bottom: true, middle: false, top: false)
+    self.barHistogram.updateDataEntries(dataEntries: dataEntries, animated: true)
+```
+
+<p align="center">
+<img src="https://raw.githubusercontent.com/Intuit/CardParts/master/images/histogram.gif" width="334" alt="Histogram" />
```
Comment (croossin) [inline README.md]: Nice gif

File: CardParts/src/Classes/Card Parts/CardPartHistogramView.swift [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            $0.trailing == self.trailingAnchor
+            $0.top == self.topAnchor
+        }
+        
+        NSLayoutConstraint.activate([
+            scrollView.heightAnchor.constraint(equalToConstant: self.intrinsicHeight)
+        ])
+    }
+    
+    /// confiogures width and spacing betweent the bar items
```
[the remark this answers is not in the document]
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartHistogramView.swift]: done

File: CardParts/src/Classes/Card Parts/CardPartHistogramView.swift (hunk 3 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartHistogramView.swift]: Fixed

File: README.md (hunk 5 above)
Comment (badrinathvm) [reply README.md]: Thank you 😊

**Pull request merged**